### PR TITLE
Add Getting Started for Next.js & Server-Side Rendering docs

### PR DESCRIPTION
**Pull request by ericclemmons**
*Issue #, if available:*
https://github.com/aws-amplify/amplify-js/issues/5435

*Description of changes:*

Screen Shot 2020-09-11 at 11 43 42
Screen Shot 2020-09-11 at 11 43 50
Screen Shot 2020-09-11 at 11 44 00
Screen Shot 2020-09-11 at 11 44 15
Screen Shot 2020-09-11 at 11 44 27
Screen Shot 2020-09-11 at 11 44 33
Screen Shot 2020-09-11 at 11 44 43

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by harrysolovay — APPROVED** [review]
LGTM

**Comment by mtliendo** [comment]
Been following this and [the discussion post](https://github.com/aws-amplify/amplify-js/issues/5435). Just wanted to say awesome work! This is super exciting!

File: docs/lib/ssr/fragments/js/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  }, [])
+
+  return ( ... )
+}
+```
+
+However, when in server-only functions (e.g. Next.js' `getServerSideProps`, `getStaticProps`, & `getStaticPaths`), use `withSSRContext({ req?: ServerRequest })`:
+
+```js
+import { Amplify, API, withSSRContext } from "aws-amplify"
```
Comment (dabit3) [inline docs/lib/ssr/fragments/js/getting-started.md:45]: Should API and Amplify still be imported here?

File: docs/lib/ssr/fragments/js/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+To work well with server-rendered pages, Amplify JS requires slight modifications from how you would use it in a client-only environment.
+
+## Amplify
+
+### Enabling SSR
```
Comment (dabit3) [inline docs/lib/ssr/fragments/js/getting-started.md:5]: Should we also mention that to make Auth SSR aware, they should use `withSSRContext` to sign in the user?

Also, is there any way to make a user authenticated via the UI components SSR aware?

**Comment by mtliendo** [comment]
On the "Getting started with SSR" page, after the first paragraph, would it be beneficial to add a paragraph/diagram/link explaining why building for SSR is different than client-side only?

That way those migrating from CRA -> NextJS for the first time have some context on why things are done differently.

File: docs/lib/ssr/fragments/js/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+### Enabling SSR 
+
+When using the Amplify CLI, the __aws-exports.js__ file gets created and updated automatically for you based upon the resources you have added and configured.
+
+For client-only apps, `Amplify.configure(awsExports)` is all you need.
+
+To enable SSR support, also provide `ssr: true`:
+
+```js
+import { Amplify } from "aws-amplify"
```
Comment (sammartinez) [inline docs/lib/ssr/fragments/js/getting-started.md:14]: Seems like we are missing semi-colons throughout. Can be a separate PR to come back through and add them. 

```suggestion
import { Amplify } from "aws-amplify";
```

File: docs/lib/ssr/fragments/js/getting-started.md (hunk 1 above)
Comment (sammartinez) [reply docs/lib/ssr/fragments/js/getting-started.md:45]: This is a far callout since we are pulling `API` from `SSR` now. I do believe the `Amplify` is there @dabit3 for the `...` of the `aws-exports`

File: docs/lib/ssr/ssr.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+title: Getting Started with SSR
```
Comment (sammartinez) [inline docs/lib/ssr/ssr.md:2]: Lets look to spell out the title

```suggestion
title: Getting Started with Server-Side Rendering (SSR)
```

File: docs/start/getting-started/nextsteps.md [hunk 5]
```diff
@@ -5,14 +5,15 @@ description: Getting Started with Amplify Framework - Next steps
...
+<inline-fragment integration="next" src="~/start/getting-started/fragments/common/nextsteps-header.md"></inline-fragment>
 <inline-fragment integration="react" src="~/start/getting-started/fragments/common/nextsteps-header.md"></inline-fragment>
 <inline-fragment integration="react-native" src="~/start/getting-started/fragments/common/nextsteps-header.md"></inline-fragment>
 <inline-fragment integration="ionic" src="~/start/getting-started/fragments/common/nextsteps-header.md"></inline-fragment>
 
 <inline-fragment integration="angular" src="~/start/getting-started/fragments/angular/nextsteps.md"></inline-fragment>
 <inline-fragment integration="js" src="~/start/getting-started/fragments/vanillajs/nextsteps.md"></inline-fragment>
 <inline-fragment integration="vue" src="~/start/getting-started/fragments/vue/nextsteps.md"></inline-fragment>
-<inline-fragment integration="react" src="~/start/getting-started/fragments/react/nextsteps.md"></inline-fragment>
+<inline-fragment integration="next" src="~/start/getting-started/fragments/react/nextsteps.md"></inline-fragment>
```
Comment (sammartinez) [inline docs/start/getting-started/nextsteps.md:16]: Is this a typo? Don't we need both `next` and `react` fragments?

File: docs/lib/ssr/fragments/js/getting-started.md (hunk 1 above)
Comment (ericclemmons) [reply docs/lib/ssr/fragments/js/getting-started.md]: Resolved via our call. Top-level imports like `API` should still be used in client-side code.

File: docs/lib/ssr/fragments/js/getting-started.md (hunk 2 above)
Comment (ericclemmons) [reply docs/lib/ssr/fragments/js/getting-started.md:5]: `withSSRContext` shouldn't be necessary on the client (and will impact bundle-sizes), but also has limited support for only `Auth`, `API`, and `DataStore` today.

UI components should be SSR-aware by default, since they defer to `Auth` that's been configured with `ssr: true`.

File: docs/start/getting-started/nextsteps.md (hunk 5 above)
Comment (ericclemmons) [reply docs/start/getting-started/nextsteps.md:16]: We do. The next steps are the same as other React apps, so I opted not to duplicate this fragment.

We don't have data or examples on if or when customers would want analtyics/storage/xr on the server, so I chose not have any callouts for that explicitly.

File: docs/start/getting-started/fragments/next/api.md [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+# Default configuration
+? How do you want users to be able to sign in?
+# Username
+? Do you want to configure advanced settings?
+# No, I am done.
+? Configure conflict detection?
+# No
+? Do you have an annotated GraphQL schema?
+# No
+? Do you want a guided schema creation?
```
Comment (dabit3) [inline docs/start/getting-started/fragments/next/api.md:44]: The CLI no longer asks this from what I can tell.

It jumps straight from "do you have an annotated GraphQL schema" to "choose a schema template" to "Do you want to edit the schema now?"

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

File: docs/start/getting-started/fragments/next/api.md (hunk 6 above)
Comment (ericclemmons) [reply docs/start/getting-started/fragments/next/api.md:44]: Weird, I can re-capture this. I took this straight from my output.


Can you paste yours for comparison?

Comment (ericclemmons) [reply docs/start/getting-started/fragments/next/api.md:44]: Ah, my version is old!

Comment (ericclemmons) [reply docs/start/getting-started/fragments/next/api.md]: Updated via 59d87759af0005affc4ebb24a9b392d0cef3700d
